sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDB.scala

```diff
@@ -1533,3 +1717,40 @@ case class AcquiredThreadInfo(
   }
 }
 
+/**
+ * A helper class to manage the lineage information when checkpoint unique id is enabled.
+ * "lineage" is an array of LineageItem (version, uniqueId) pair.
+ *
+ * The first item of "lineage" should normally be the version of a snapshot, except
+ * for the first few versions. Because they are solely loaded from changelog file.
+ * (i.e. with default minDeltasForSnapshot, there is only 1_uuid1.changelog, no 1_uuid1.zip)
+ *
+ * The last item of "lineage" corresponds to one version before the to-be-committed version.
```

Comment:
Why do we need to have a non-committed version in the lineage? Can you remind me at which scenario we need to deal with an uncommitted lineage item?

Reply:
ah no this means lineage only contains committed versions. It basically means we only append lineage item during commit: 
https://github.com/apache/spark/blob/91d5e815175b32275b34922790d9bb4ba5208e1f/sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDB.scala#L866

Reply:
Got it. Thanks for your explanation.